src/-private/services/data-key-map.ts

```diff
@@ -0,0 +1,3 @@
+import KeyMapFactory from '../factories/key-map-factory.ts';
```

Comment:
why not just define the services in this file instead of in a "factories" folder?

Reply:
Fair point, I moved them.